workflows/sars-cov-2-variant-calling/sars-cov-2-pe-illumina-artic-ivar-analysis/pe-wgs-ivar-analysis.ga

```diff
@@ -0,0 +1,1215 @@
+            "tool_version": "2.0.2+galaxy2",
+            "type": "tool",
+            "uuid": "217c653d-84a7-49dd-ba0f-601bb651ee7d",
+            "workflow_outputs": []
+        },
+        "9": {
+            "annotation": "",
+            "content_id": "toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.3.1+galaxy0",
+            "errors": null,
+            "id": 9,
+            "input_connections": {
+                "input_bam": {
+                    "id": 7,
+                    "output_name": "output1"
+                },
+                "primer|input_bed": {
+                    "id": 0,
+                    "output_name": "output"
+                }
+            },
+            "inputs": [],
+            "label": null,
+            "name": "ivar trim",
+            "outputs": [
+                {
+                    "name": "output_bam",
+                    "type": "bam"
+                }
+            ],
+            "position": {
+                "bottom": 146.13333129882812,
+                "height": 184.8000030517578,
+                "left": 196.75,
+                "right": 396.75,
+                "top": -38.66667175292969,
+                "width": 200,
+                "x": 196.75,
+                "y": -38.66667175292969
+            },
+            "post_job_actions": {
+                "HideDatasetActionoutput_bam": {
+                    "action_arguments": {},
+                    "action_type": "HideDatasetAction",
+                    "output_name": "output_bam"
+                }
+            },
+            "tool_id": "toolshed.g2.bx.psu.edu/repos/iuc/ivar_trim/ivar_trim/1.3.1+galaxy0",
+            "tool_shed_repository": {
+                "changeset_revision": "cf65217ad61c",
+                "name": "ivar_trim",
+                "owner": "iuc",
+                "tool_shed": "toolshed.g2.bx.psu.edu"
+            },
+            "tool_state": "{\"amplicons\": {\"filter_by\": \"\", \"__current_case__\": 1}, \"inc_primers\": \"true\", \"input_bam\": {\"__class__\": \"ConnectedValue\"}, \"min_len\": \"30\", \"min_qual\": \"20\", \"primer\": {\"source\": \"history\", \"__current_case__\": 0, \"input_bed\": {\"__class__\": \"ConnectedValue\"}}, \"primer_pos_wiggle\": \"0\", \"window_width\": \"4\", \"__page__\": null, \"__rerun_remap_job_id__\": null}",
```

Comment:
You may want to consider "filter_by": "yes" here to make use of ivar trim's new `-f` option, which makes use of knowledge about expected amplicons to discard reads that do not fall within amplicon boundaries.

The downside is that you'll require one more input dataset, but it seems to be worth it -> https://github.com/andersen-lab/ivar/issues/75

Reply:
This is one of the improvements in v0.3 of (https://github.com/galaxyproject/iwc/tree/main/workflows/sars-cov-2-variant-calling/sars-cov-2-pe-illumina-artic-variant-calling)